tests/StereoDB.Tests/HeartbeatTests.fs

```diff
@@ -0,0 +1,51 @@
+module Tests.HeartbeatTests
+
+open System.Collections.Concurrent
+open System.Threading.Tasks
+
+open Swensen.Unquote
+open Xunit
+
+open Confluent.Kafka
+
+open StereoDB
+open StereoDB.FSharp
+open Tests.TestHelper
+open System
+open System.Threading
+
+[<Fact>]
+[<Trait("CI", "disable")>]
+let ``Heartbeat is sent on db init`` () = task {
+    let settings = StereoDbSettings.OnlyInMemory
+
+    use! db = StereoDb.init(Schema(), settings)
+    use kafkaConsumer = ConsumerBuilder<string, string>(
+        ConsumerConfig(
+            BootstrapServers = settings.HeartbeatConfig.KafkaBootstrapServers,
+            AutoOffsetReset = AutoOffsetReset.Latest,
+            GroupId = Random.Shared.NextInt64().ToString())).Build()
+
+    use cts = new CancellationTokenSource()
+    cts.CancelAfter(TimeSpan.FromSeconds(10)) // wait for heartbeat messages to be sent
+
+    let tcs = TaskCompletionSource()
+    let heartbeatMessages = ConcurrentBag<string>()
+
+    task {
+        kafkaConsumer.Subscribe(settings.HeartbeatConfig.KafkaHeartbeatTopic(settings.ClusterId))
+        try
+            while not cts.IsCancellationRequested do
+                do! Task.Yield()
```

Comment:
why do you need Task Yielding for each iteration?
It seems you can have use only once, on top of while

```
do! Task.Yield()
try
    while not cts.IsCancellationRequested do
```

Reply:
good catch, updated

Reply:
agree, removed, that was initally added to resolve issue with blocking producer and consumer